src/main/java/org/dcsa/core/exception/handler/GlobalExceptionHandler.java

```diff
@@ -24,42 +25,66 @@
 public class GlobalExceptionHandler {
 
   @ExceptionHandler(ConcreteRequestErrorMessageException.class)
-  public ResponseEntity<RequestFailureTO> handle(ServerHttpRequest serverHttpRequest, ConcreteRequestErrorMessageException ex) {
+  public ResponseEntity<RequestFailureTO> handle(
+      ServerHttpRequest serverHttpRequest, ConcreteRequestErrorMessageException ex) {
     ResponseStatus responseStatusAnnotation = ex.getClass().getAnnotation(ResponseStatus.class);
     HttpStatus httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
     ConcreteRequestErrorMessageTO errorEntity = ex.asConcreteRequestMessage();
     if (responseStatusAnnotation != null) {
       httpStatus = responseStatusAnnotation.value();
     }
-    RequestFailureTO failureTO = new RequestFailureTO(
+    RequestFailureTO failureTO =
+        new RequestFailureTO(
             serverHttpRequest.getMethodValue(),
             serverHttpRequest.getURI().toString(),
             List.of(errorEntity),
-            httpStatus
-    );
+            httpStatus);
     return new ResponseEntity<>(failureTO, httpStatus);
   }
 
   @ExceptionHandler(DCSAException.class)
-  public void handleDCSAExceptions(DCSAException dcsaEx) {
+  public ResponseEntity<RequestFailureTO> handleDCSAExceptions(
+      ServerHttpRequest serverHttpRequest, DCSAException dcsaEx) {
+
     log.debug(
         "{} ({}) - {}",
         this.getClass().getSimpleName(),
         dcsaEx.getClass().getSimpleName(),
         dcsaEx.getMessage());
     logExceptionTraceIfEnabled(dcsaEx);
-    throw dcsaEx;
+
+    ResponseStatus responseStatusAnnotation = dcsaEx.getClass().getAnnotation(ResponseStatus.class);
+
+    HttpStatus httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
+
+    if (responseStatusAnnotation != null) {
+      httpStatus = responseStatusAnnotation.value();
+    }
+
+    ConcreteRequestErrorMessageTO errorEntity =
+        new ConcreteRequestErrorMessageTO("internal_error", dcsaEx.getMessage());
+
+    RequestFailureTO failureTO =
+        new RequestFailureTO(
+            serverHttpRequest.getMethodValue(),
+            serverHttpRequest.getURI().toString(),
+            List.of(errorEntity),
+            httpStatus);
+    return new ResponseEntity<>(failureTO, httpStatus);
   }
 
-  @ResponseStatus(value = HttpStatus.BAD_REQUEST, reason = "Invalid input.")
+  @ResponseStatus(value = HttpStatus.BAD_REQUEST, reason = "invalid_input.")
```

Comment:
Should this be camelCase instead as well or will it be the free text field?  (If it is free text value, then I prefer the original version).